Engine/HistoricalData/SubscriptionDataReaderHistoryProvider.cs

```diff
@@ -79,6 +79,11 @@ public override IEnumerable<Slice> GetHistory(IEnumerable<HistoryRequest> reques
         /// </summary>
         protected virtual Subscription CreateSubscription(HistoryRequest request, DateTime start, DateTime end)
         {
+            // Tradable dates are defined with the data time zone to access the right source
+            var tradableDates = Time.EachTradeableDay(request.ExchangeHours,
```

Comment:
Believe we should use `EachTradeableDayInTimeZone` the same as the data subscription stack, see https://github.com/QuantConnect/Lean/blob/master/Common/Data/UniverseSelection/SubscriptionRequest.cs#L77, which checks the exchange is open in exchange tz, versus here we will be checking the exchange is open in data tz.